local-volume/provisioner/pkg/common/common.go

```
@@ -167,28 +161,57 @@ func GetDefaultVolumeConfig() map[string]MountConfig {
-		}
-		configMapData[class] = string(val)
+	val, err := yaml.Marshal(config.StorageClassConfig)
+	if err != nil {
+		return nil, fmt.Errorf("unable to Marshal volume config: %v", err)
 	}
+	configMapData[ProvisonerStorageClassConfig] = string(val)
 	return configMapData, nil
 }
 
 // ConfigMapDataToVolumeConfig converts configmap data to volume config
-func ConfigMapDataToVolumeConfig(data map[string]string) (map[string]MountConfig, error) {
-	mountConfig := make(map[string]MountConfig)
-	for class, val := range data {
-		config := MountConfig{}
-		if err := json.Unmarshal([]byte(val), &config); err != nil {
-			return nil, fmt.Errorf("unable to unmarshal config for class %v: %v", class, err)
+func ConfigMapDataToVolumeConfig(data map[string]string, provisionerConfig *ProvisionerConfiguration) error {
+	rawYaml := ""
+	for key, val := range data {
+		rawYaml += key
+		rawYaml += ": \n"
+		rawYaml += insertSpaces(string(val))
+	}
+	if err := yaml.Unmarshal([]byte(rawYaml), provisionerConfig); err != nil {
+		return fmt.Errorf("fail to Unmarshal yaml due to: %#v", err)
+	}
+	return nil
+}
+
+func insertSpaces(original string) string {
+	spaced := ""
+	for _, line := range strings.Split(original, "\n") {
+		spaced += "   "
+		spaced += line
+		spaced += "\n"
+	}
+	return spaced
+}
+
+// LoadProvisionerConfigs loads all configuration into a string and unmarshal it into ProvisionerConfiguration struct.
+// The configuration is stored in the configmap which is mounted as a volume.
+func LoadProvisionerConfigs(provisionerConfig *ProvisionerConfiguration) error {
+	files, err := ioutil.ReadDir(ProvisionerConfigPath)
+	if err != nil {
+		return err
+	}
+	data := make(map[string]string)
+	for _, file := range files {
+		if !file.IsDir() {
+			fileContents, err := ioutil.ReadFile(path.Join(ProvisionerConfigPath, file.Name()))
+			if err != nil {
+				glog.Infof("Could not read file: %s due to: %v", path.Join(ProvisionerConfigPath, file.Name()), err)
+				continue
```

I would fail instead of continuing.  We don't want to continue and potentially ignore user configuration.

---

Cannot fail because there is one file which is always return ..data in the list which cannot be read and I feel hardcoding to skip it is not ideal.